### PR TITLE
Modify functional tests to run on Gateway

### PR DESCRIPTION
**Pull request by kannappanr**
Add 'NotImplemented' error handlers,
Add wait time before recreating a bucket in Azure,

### CODE REVIEW

**Comment by deekoder** [comment]
Need this fix for release. So marking this high pri.

File: src/test/functional/functional-tests.js [hunk 1]
```diff
@@ -1085,9 +1105,14 @@ describe('functional tests', function() {
  
       // This test is very similar to that above, except it does not include
       // Minio.ObjectCreatedAll in the config. Thus, no events should be emitted.
-      step('listenBucketNotification(bucketName, prefix, suffix, events)_events:ObjectRemoved_', done => {
+      step(`listenBucketNotification(bucketName, prefix, suffix, events)_bucketName:${bucketName}, events:s3:ObjectRemoved:*`, done => {
```
Comment (harshavardhana) [inline src/test/functional/functional-tests.js]: Does making minio-js functional test work with GCS requires all of this cleanup? @kannappanr can we do it in another PR?

**Comment by kannappanr** [comment]
Pushed again, based on the review comments. Please review again.

**Comment by harshavardhana** [comment]
Build is failing @kannappanr

File: src/test/functional/functional-tests.js [hunk 2]
```diff
@@ -622,27 +629,35 @@ describe('functional tests', function() {
...
     let policies = [Policy.READONLY, Policy.WRITEONLY, Policy.READWRITE]
 
     // Iterate through the basic policies ensuring it can set and check each of them.
     policies.forEach(policy => {
+      policyNotImplemented = false
       step(`setBucketPolicy(bucketName, objectPrefix, bucketPolicy, cb)_bucketPolicy:${policy}_`, done => {
         client.setBucketPolicy(bucketName, '', policy, err => {
-          if (err) return done(err)
+          if (err) {
+            if(err.code != 'NotImplemented') done(err)
```
Comment (ebozduman) [inline src/test/functional/functional-tests.js]: Please make sure `status = 'NA'` and `alert = 'Not Implemented'` without having any error message in `message` and any stack trace in `error` fields in the logs.

Comment (kannappanr) [reply src/test/functional/functional-tests.js]: There won't be any error and stack messages. Currently, there is no way to pass status 'NA' to the repoter. Will file a separate issue for it and work on it there.
